src/core/search/search.cc

```diff
@@ -79,12 +79,17 @@ struct BasicSearch {
 SearchAlgorithm::SearchAlgorithm() = default;
 SearchAlgorithm::~SearchAlgorithm() = default;
 
-SearchAlgorithm::SearchAlgorithm(string_view query)
-    : query{make_unique<AstExpr>(ParseQuery(query))} {
+bool SearchAlgorithm::Init(string_view query) {
+  try {
+    query_ = make_unique<AstExpr>(ParseQuery(query));
```

Comment:
make a comment explaining why do you try/catch.
catch the exception thrown by parser, ideally vlog it as well.